File: articles/donald-knuth-and-the-art-of-computer-programming/c/growth_comparison.c
#include <math.h>
#include <stdio.h>

int main(void) {
    double n = 1000.0;
    printf("log2_n=%.6f\n", log2(n));
    printf("n=%.0f\n", n);
    printf("n_log2_n=%.6f\n", n * log2(n));
    printf("n_squared=%.0f\n", n * n);
    return 0;
}

File: articles/donald-knuth-and-the-art-of-computer-programming/go/growth_comparison.go
package main

import (
	"fmt"
	"math"
)

func main() {
	n := 1000.0
	fmt.Printf("log2_n=%.6f\n", math.Log2(n))
	fmt.Printf("n=%.0f\n", n)
	fmt.Printf("n_log2_n=%.6f\n", n*math.Log2(n))
	fmt.Printf("n_squared=%.0f\n", n*n)
}
